### PR TITLE
Checking style prop height and width in MapGL's componentDidUpdate

### LINKED ISSUE #272: Unpredictable resizing behaviour
First of all, let me thank you for this library. We just switched to it from `uber/react-map-gl` and are much happier with it. In particular, the cluster and inertia panning functionalities are great.

One thing that we are still struggling with is the auto-sizing functionality, which worked better with `uber`. We are using `<Autosizer>` from the https://github.com/bvaughn/react-virtualized library to dynamically adjust the map for responsive designs.

Our code looks basically like this:

```js

<AutoSizer>
            {({ height, width }) => {
              return (
                  <MapGL
                    ref={this.map}
                    style={{ width, height }}
                    {...this.state.viewport}
                    mapStyle={`https://api.maptiler.com/maps/basic/style.json?key=${key}`}
                    onViewportChange={this.onViewportChange}
                    viewportChangeMethod={viewportChangeMethod}
                    viewportChangeOptions={viewPortChangeOptions}
                    onClick={onClick}
                  >
            );
      }}
</AutoSizer>
```

With this code, the map sometimes gets very small (or isn't even visible), sometimes it's just a bit too small.
We've noticed that `<AutoSizer>` sometimes reports wrong sizes for a moment (like height or width being 0), but it catches up and then uses the correct sizes. It seems as if your library cannot keep up with this. It looks as if some debouncing is happening inside the library, but it doesn't work well.

We added some safeguarding code that checks for width or height being 0 and re-using the previous sizes instead. This helped a bit, but did not completely solve our problem.

The problem is most prominent if you toggle Chrome's developer tools on and off. Without the safe-guarding code, sometimes the map disappears completely, or it is only 20px high.

### PR DESCRIPTION
**Pull request by cazzer**
This should resolve #272, but I have some questions/issues running tests.

The biggest issue is when calling `this._map.resize()` during MapGL's `componentDidUpdate` hook: when running in a browser environment this runs as expected, but when running tests it throws the following error:
```
TypeError: this._map.resize is not a function

      520 |     }
      521 | 
    > 522 |     this._map.resize();
          |               ^
      523 |   }
      524 | 
      525 |   /**
```
I thought this may be due to how I'm mounting/rendering the component with Enzyme, but am not super familiar with the tool and couldn't find any hints after a cursory search. Guidance would be appreciated.

There are also ~two~ one ESLint issue preventing me from committing without the `--no-verify` flag:
```
/Users/calebbrewer/dev/cazzer/react-map-gl/src/components/Marker/index.js
  8:32  error  '@mapbox/point-geometry' should be listed in the project's dependencies. Run 'npm i -S @mapbox/point-geometry' to add it  import/no-extraneous-dependencies
```

### CODE REVIEW

File: src/components/MapGL/index.test.js [hunk 1]
```diff
@@ -332,3 +332,48 @@ test('renders without mapbox-gl', () => {
...
+    }
+  }
+
+  const wrapper = mount(<Wrapper />);
+
+  const map = wrapper.instance().mapRef.current.getMap();
+  const resizeSpy = jest.fn();
+
+  map.on('resize', () => {
+    if (wrapper.children().props().style.height === resizeHeight) {
```
Comment (cazzer) [inline src/components/MapGL/index.test.js:369]: This feels like a weird way to verify that the container has resized appropriately, but I wasn't able to `wrapper.find('canvas')` or any other elements that might reflect the new size. Does anyone have thoughts on an alternative approach or does this seem sufficient?

File: src/components/MapGL/index.js [hunk 2]
```diff
@@ -502,6 +503,25 @@ class MapGL extends PureComponent<Props, State> {
...
+  _updateMapSize(prevProps: Props, newProps: Props): void {
+    const sizeChanged =
+      prevProps.style?.height !== newProps.style?.height ||
+      prevProps.style?.width !== newProps.style?.width;
+
+    if (!sizeChanged) {
+      return;
+    }
+
+    this._map.resize();
```
Comment (cazzer) [inline src/components/MapGL/index.js:522]: @stepankuzmin just wanted to make sure you saw my note about this. During tests, this call will actually fail despite working fine in a real browser. Any thoughts on why that is and how I could get the test working?

**Comment by cazzer** [comment]
I updated the test to mock `map.resize` which resolved my issue. I also added `@mapbox/point-geometry` since it was causing issues while committing. It should be good to go now!

File: package.json [hunk 3]
```diff
@@ -143,5 +143,8 @@
         "dist/react-map-gl.esm.js"
       ]
     }
-  ]
+  ],
+  "dependencies": {
+    "@mapbox/point-geometry": "^0.1.0"
```
Comment (stepankuzmin) [inline package.json:148]: Hi @cazzer! What was the issue that needed to add `@mapbox/point-geometry` as a dependency?

Comment (cazzer) [reply package.json:148]: I got the following ESLint error when trying to commit:
```
/Users/calebbrewer/dev/cazzer/react-map-gl/src/components/Marker/index.js
  8:32  error  '@mapbox/point-geometry' should be listed in the project's dependencies. Run 'npm i -S @mapbox/point-geometry' to add it  import/no-extraneous-dependencies
```
I can remove it, but the build will fail.

**Comment by cazzer** [comment]
@stepankuzmin let me know if there is anything you'd like me to update in this PR, I'd love to get the change in!

**Comment by cazzer** [comment]
@stepankuzmin I dropped the dependency from this PR in case that was holding up the review.

**Pull request merged**